app/src/main/java/com/example/euriskocodechallenge/ui/home/news/NewsFragment.kt

```diff
@@ -44,15 +46,14 @@ class NewsFragment : Fragment() {
     }
 
     private fun checkConnection(view: View) {
-        connectionStatus = ConnectivityLiveData(requireActivity().application)
-        connectionStatus.observe(viewLifecycleOwner) {
+        ConnectivityLiveData(requireActivity().application).observe(viewLifecycleOwner) {
             if (it) {
                 binding.internetTv.visibility = View.GONE
                 binding.newsProgress.visibility = View.VISIBLE
                 initRecyclerView()
                 observeViewModel(view)
             } else if (!it) {
```

Comment:
@martin371998 not sure why you're checking for negation here, simple if/else would be enough

Reply:
Done